python/ml4ir/base/model/optimizer.py

```diff
@@ -37,38 +24,83 @@ def get_optimizer(
     References:
         https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Optimizer
         https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/schedules/ExponentialDecay
+        https://arxiv.org/pdf/1506.01186.pdf
 
     FIXME:
         Define all arguments overriding tensorflow defaults in a separate file
         for visibility with ml4ir users
     """
-    # Define an exponential learning rate decay schedule
-    learning_rate_schedule = ExponentialDecay(
-        learning_rate,
-        decay_steps=learning_rate_decay_steps,
-        decay_rate=learning_rate_decay,
-        staircase=True,
-    )
+    model_config = file_io.read_yaml(model_config_file)
 
-    if optimizer_key == OptimizerKey.ADAM:
-        return tf_optimizers.Adam(
-            learning_rate=learning_rate_schedule, clipvalue=gradient_clip_value
-        )
-    elif optimizer_key == OptimizerKey.NADAM:
-        return tf_optimizers.Nadam(
-            learning_rate=learning_rate_schedule, clipvalue=gradient_clip_value
-        )
-    elif optimizer_key == OptimizerKey.ADAGRAD:
-        return tf_optimizers.Adagrad(
-            learning_rate=learning_rate_schedule, clipvalue=gradient_clip_value
-        )
-    elif optimizer_key == OptimizerKey.SGD:
-        return tf_optimizers.SGD(
-            learning_rate=learning_rate_schedule, clipvalue=gradient_clip_value
-        )
-    elif optimizer_key == OptimizerKey.RMS_PROP:
-        return tf_optimizers.RMSprop(
-            learning_rate=learning_rate_schedule, clipvalue=gradient_clip_value
+    if 'lr_schedule' not in model_config:
+        learning_rate_schedule = ExponentialDecay(
+            initial_learning_rate=0.01,
+            decay_steps=10000000,
+            decay_rate=1.0,
+            staircase=True,
         )
+
+    else:
```

Comment:
@mohazahran We shouldn't have the defaults in the code IMO.
Option 1: Fallback to tf defaults
Option 2: Have them defined in a separate constants class above for the time being perhaps?

Reply:
@balikasg In case you have any other ideas?

Reply:
I prefer the TF defaults. Unless the user explicitly asks for something different, I think it makes sense to keep the defaults. It is not tricky to implement with the distinction we have now in model config. Just initialize and pass the dictionary of the user params as `**kwags`.  The other advantage of doing this is pushing the arg validation to the user and tf.